src/python/tests/test_statistics.py

```diff
@@ -0,0 +1,143 @@
+        event_loop.run_until_complete(set_in_keys())
+
+    def teardown_method(self):
+        """Set up each test with fresh data, and empty _KEYS dictionary"""
+        _KEYS.clear()
+        os.system("psql < `locate foglamp_ddl.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+        os.system("psql < `locate foglamp_init_data.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+
+    async def test_update_value(self):
+        """
+        Test that statistics table gets updated
+        :assert:
+            1. value gets update with new rand_value
+            2. previous value does not change
+        """
+        for key in _KEYS:
+            rand_value = random.randint(1, 10)
+            await update_statistics_value(statistics_key=key, value_increment=rand_value)
+            stmt = sa.select([_STATISTICS_TBL.c.value]).select_from(_STATISTICS_TBL).where(
+                _STATISTICS_TBL.c.key == key)
+            try:
+                async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                    async with engine.acquire() as conn:
+                        async for result in conn.execute(stmt):
+                            assert result[0] == rand_value
+            except Exception:
+                raise
+
+            stmt = sa.select([_STATISTICS_TBL.c.previous_value]).select_from(_STATISTICS_TBL).where(
+                _STATISTICS_TBL.c.key == key)
+            try:
+                async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                    async with engine.acquire() as conn:
+                        async for result in conn.execute(stmt):
+                            assert result[0] == 0
+            except Exception:
+                raise
+
+    async def test_update_consecutive_value(self):
+        """
+        Multiple Insert into table
+        :assert:
+            Assert that the value in statistics table is equal to the sum of value_increments
+        """
+        for key in _KEYS:
+            rand_value = [random.randint(1, 10), random.randint(1, 10)]
+            await update_statistics_value(statistics_key=key, value_increment=rand_value[0])
+            await update_statistics_value(statistics_key=key, value_increment=rand_value[1])
+
+            stmt = sa.select([_STATISTICS_TBL.c.value,
+                              _STATISTICS_TBL.c.previous_value]
+                            ).select_from(_STATISTICS_TBL).where(
+                                _STATISTICS_TBL.c.key == key)
+            try:
+                async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                    async with engine.acquire() as conn:
+                        async for result in conn.execute(stmt):
+                            assert result[0] == sum(rand_value)
+                            assert result[1] == 0
+            except Exception:
+                raise
+
+    async def test_invalid_statistics_key(self):
```

Comment:
looks invalid test to me!

Reply:
@praveen-garg the test is invalid... That said, I'm wondering if an error should be returned when trying to update a statistics value (`update_statistcs_value`) with `statistics_key` that does not exist? Right now when doing such a thing, then it's as if nothing happened. 
CC @ashwinscale @MarkRiddoch